ports/zephyr/mpconfigport.h

```diff
@@ -139,3 +142,20 @@ typedef long mp_off_t;
 // extra built in names to add to the global namespace
 #define MICROPY_PORT_BUILTINS \
     { MP_ROM_QSTR(MP_QSTR_open), MP_ROM_PTR(&mp_builtin_open_obj) },
+
+#if MICROPY_PY_THREAD
+#define MICROPY_EVENT_POLL_HOOK \
+    do { \
+        extern void mp_handle_pending(bool); \
+        mp_handle_pending(true); \
+        MP_THREAD_GIL_EXIT(); \
+        MP_THREAD_GIL_ENTER(); \
```

Comment:
Maybe this should do a `k_msleep(1)` between GIL_EXIT and GIL_ENTER, otherwise things like `mp_hal_stdin_rx_chr()` become busy loops with no chance to power-save the MCU.